data/transactions/logic/assembler.go

```diff
@@ -1101,6 +1101,63 @@ func typeDig(ops *OpStream, args []string) (StackTypes, StackTypes) {
 	return anys, returns
 }
 
+func typeCover(ops *OpStream, args []string) (StackTypes, StackTypes) {
+	if len(args) == 0 {
+		return oneAny, oneAny
+	}
+	n, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return oneAny, oneAny
+	}
+	depth := int(n) + 1
+	anys := make(StackTypes, depth)
+	for i := range anys {
+		anys[i] = StackAny
+	}
+	returns := make(StackTypes, depth)
+	for i := range returns {
+		returns[i] = StackAny
+	}
+	idx := len(ops.typeStack) - depth
+	if idx >= 0 {
+		sv := ops.typeStack[len(ops.typeStack)-1]
+		for i := idx; i < len(ops.typeStack)-1; i++ {
```

Comment:
Could this be shifted more cleanly using `copy()`?

Reply:
It could, but I'm to blame for setting the precedent with `dig`. We need cleaner helper functions for building these arrays in the type* functions.  Then we can also do `select`, `dup`, `dup`.

Reply:
Agreed. Does it make sense to include those changes in this PR or should it be done in a separate one?

Reply:
I think we should do those separately.  I'm still mulling over the right interface, but I think I'm imagining two things, one I'm fairly sure about, and one less so.

Fairly sure: the typeCheck functions should check the appropriate depth first, and return early if the stack is short (with an appropriately sized slice to cause checkStack to complain about length).  This means the rest of the typeCheck function can assume the stack is deep enough, and it'll be easier to write the code (using copy!) that returns slices that match parts of the existing stack.

Less sure: the interface should be reworked a little so that immediates are parsed early in one place, and supplied in a structured way to the typeCheck and assemble functions.  That way it only happens once, and assemble functions are no longer expected to do that parsing or error checking.

Reply:
I agree with the first one for sure, the second point is good as well - would that still mean if a program failed in typeCheck it fails during assembly?